### DISCUSSION TITLE
Rules cli stops if a snake returns invalid response

### DISCUSSION THREAD

**Discussion by jlafayette** [BattlesnakeOfficial/feedback#205] [Feature Requests]
Currently if a snake returns a response with invalid JSON, the game will exit immediately with an error like this:
```
2022/07/26 04:57:19 invalid character '<' looking for beginning of value
```

I'm hoping that we could match the behavior of the official game engine in these cases and have the snake continue in the direction of it's last move.

Also I think it would be helpful to show a warning message with more details to help debugging - including the body would be especially helpful, since that often contains an error message.  For example, this is an out of bounds error from biter:
```
2022/07/26 05:34:28 [WARN]: Got non-ok status code from http://localhost:5001/move
        StatusCode: 502 (expected 200)
        Body: "<html xmlns=\"http://www.w3.org/1999/xhtml\"><head><title>Jester route exception</title></head><body><h1>An error has occured in one of your routes.</h1><p><b>Detail: </b>fatal.nim(53)            sysFatal<br/>\nasyncfutures.nim(389)    read<br/>\nasyncfutures.nim(389)    read<br/>\nindex -1 not in 0 .. 76</p></body></html>"
```

In cases where the StatusCode is 200, but the body is not a valid JSON, it could show something like this:
```
2022/07/26 05:33:19 [WARN]: Failed to decode JSON from http://localhost:5000/move
        Error: invalid character 'u' looking for beginning of value
        Body: "up"
        See https://docs.battlesnake.com/references/api#post-move
```

If the JSON is valid but the move is not, then it could show what the valid options are:
```
2022/07/26 05:32:14 [WARN]: Failed to parse JSON from http://localhost:5000/move
        Error: invalid move "forward", valid moves are "up", "down", "left" or "right"
        Body: "{\n  \"move\": \"forward\"\n}\n"
        See https://docs.battlesnake.com/references/api#post-move
```

**Comment by bvanvugt** [comment]
I like this idea, thanks for detailing it. For reference, I believe the relevant code is here:
https://github.com/BattlesnakeOfficial/rules/blob/main/cli/commands/play.go#L412

Interestingly I don't think there's actually a check for 200 OK...

**Comment by jlafayette** [comment]
I opened a PR to address this: https://github.com/BattlesnakeOfficial/rules/pull/95

> Interestingly I don't think there's actually a check for 200 OK...

I noticed that too... with my proposed change it checks for 200 status code and does some other validation on the JSON and move string.

**Reply by bvanvugt** [reply]
Looks great, merged! Thanks @jlafayette :)
